### PR TITLE
Exercise 2 Commit

### PR DESCRIPTION
**Pull request by konradkaniewskiey**

### CODE REVIEW

File: src/DecileCounter.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
+import java.util.*;
+
+public class DecileCounter {
+    public static void Decile() {
```
Comment (jaroslawwoloch) [inline src/DecileCounter.java:4]: why static?

File: src/DecileCounter.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        Scanner inputScanner = new Scanner(System.in);
+        List<Integer> numbers = new ArrayList<>();
+        int num;
+        do {
+            System.out.print("Enter a number from 1 to 1000 (or 'END' to stop): ");
```
Comment (jaroslawwoloch) [inline src/DecileCounter.java:9]: .... enter a number and hit RETURN ...... - it's needed to finish line and trigger scaner

File: src/DecileCounter.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+            String str = inputScanner.nextLine();
+            if (str.equals("END")) {
+                break;
+            }
+            try {
+                num = Integer.parseInt(str);
+                if (num < 1 || num > 1000) {
+                    System.out.println("Invalid input. Number must be between 1 and 1000.");
```
Comment (jaroslawwoloch) [inline src/DecileCounter.java:17]: on this sytuation You have message :Enter...... then number provided by user and error in this same line - put error message into new line

File: src/DecileCounter.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        } while (true);
+
+        Collections.sort(numbers);
+        int size = numbers.size();
+        int thirdDecileIndex = (int) Math.ceil(size * 0.3) - 1;
+        int sixthDecileIndex = (int) Math.ceil(size * 0.6) - 1;
+        int thirdDecileCount = 0;
+        int sixthDecileCount = 0;
+
+        for (int i = 0; i < size; i++) {
```
Comment (jaroslawwoloch) [inline src/DecileCounter.java:33]: this don't meet business requirements. Read what is decile https://en.wikipedia.org/wiki/Decile and set for all population (1-1000)

**Review by jaroslawwoloch — CHANGES_REQUESTED** [review]

File: src/DecileCounter.java (hunk 3 above)
Comment (konradkaniewskiey) [reply src/DecileCounter.java:17]: image

Not sure what you mean, looks fine to me.

Comment (jaroslawwoloch) [reply src/DecileCounter.java:17]: ok Scanner.nextLine after read move cursor to next line but other scanner.next... methods don't do that.

**Pull request merged**